lib/provisioning/Provisioner.js

```diff
@@ -128,60 +129,48 @@ function Provisioner(ircBridge, enabled, requestTimeoutSeconds) {
         next();
     });
 
-    as.app.post("/_matrix/provision/link", Promise.coroutine(function*(req, res) {
-        try {
-            yield self.requestLink(req.body);
-            res.json({});
-        }
-        catch (err) {
-            res.status(500).json({error: err.message});
-            throw err;
-        }
-    }));
+    let createProvisionEndpoint = (fn, fnName) => {
+        return Promise.coroutine(function*(req, res) {
+            req = new ProvisionRequest(req, fnName);
+            req.log.info(
+                'New provisioning request: ' + JSON.stringify(req.body) +
+                ' params: ' + JSON.stringify(req.params)
+            );
+            try {
+                let result = yield fn.apply(self, [req]);
```

Comment:
Also I would be tempted to make `fn` be a bound function so you don't need to do this (which means you can yank `createProvisionEndpoint` out to be a stand-alone function without referencing `self`), but I don't really mind either way.